src/SideNavigation/SideNavigation.js

```diff
@@ -1,204 +1,263 @@
+                        children: (<React.Fragment>
+                            {glyph ? (
+                                <span
+                                    className={`fd-side-nav__icon sap-icon--${glyph} sap-icon--l`}
+                                    role='presentation' />
+                            ) : null}
+                            {child.props.children}
+                        </React.Fragment>),
+                        className: getClasses()
+                    });
+                } else if (React.isValidElement(child) && child.type === SideNavSubList) {
+                    return React.cloneElement(child, {
+                        onItemSelect: onItemSelect,
+                        open: expandedIds.includes(id),
+                        selectedId: selectedId
+                    });
+                } else {
+                    return child;
+                }
+            })}
+        </li>
     );
+};
 
+SideNavListItem.propTypes = {
+    children: PropTypes.node,
+    expandedIds: PropTypes.array,
+    glyph: PropTypes.string,
+    id: PropTypes.string,
+    isSubItem: PropTypes.bool,
+    name: PropTypes.string,
+    selected: PropTypes.bool,
+    selectedId: PropTypes.string,
+    url: PropTypes.string,
+    onClick: PropTypes.func,
+    onItemSelect: PropTypes.func
+};
+
+SideNavListItem.propDescriptions = {
+    expandedIds: 'An array of strings for expanded `SideNavListItems`\'s.',
+    isSubItem: 'Set to **true** to display as a subitem.',
+    name: 'Link text to be set in conjunction with the `url` prop.',
+    onItemSelect: 'Callback function when user selects an item .',
+    selected: 'Set to **true** to display as selected.',
+    selectedId: 'The id of the selected `SideNavListItem`.',
+    url: 'Creates an internal anchor when a child anchor is not provided.'
+};
+
+SideNavListItem.defaultProps = {
+    onClick: () => {}
+};
+
+export const SideNavSubList = ({children, id, onItemSelect, open, selectedId, ...props}) => {
     return (
-        <div
-            {...rest}
-            className={sideNavGroupClasses}>
-            <h1 {...titleProps} className='fd-side-nav__title'>{title}</h1>
-            {children}
-        </div>
+        <ul
+            {...props}
+            aria-expanded={open}
+            aria-hidden={!open}
+            className='fd-side-nav__sublist'
+            key={id}>
+            {
+                React.Children.map(children, (child) => {
+                    return React.cloneElement(child, {
+                        isSubItem: true,
+                        onItemSelect: onItemSelect,
+                        selected: selectedId === child.props.id
+                    });
+                })
+            }
+        </ul>
     );
 };
 
-SideNavGroup.propTypes = {
-    title: PropTypes.string.isRequired,
-    className: PropTypes.string,
-    titleProps: PropTypes.object
+SideNavSubList.propTypes = {
+    children: PropTypes.node,
```

Comment:
Same comment as above regarding specific instances of children.

Reply:
After trying to tackle this, there is not a great way in React to enforce what components are passed as children. For now we are ignoring this issue per a conversation with @greg-a-smith.